test/gtest/uct/test_flush.cc

```diff
@@ -538,3 +539,255 @@ UCS_TEST_SKIP_COND_P(uct_flush_test, am_pending_flush_nb,
+        }
+
+        volatile bool flag;
+        static void comp_cb(uct_completion_t *self, ucs_status_t status) {
+            comp *c = static_cast<comp*>(self);
+            ASSERT_UCS_OK(status);
+            c->flag = true;
+        }
+    };
+
+    void flush_and_reconnect() {
+        std::list<entity *> flushing;
+        ucs_status_t status = UCS_OK;
+        struct comp done;
+
+        flushing.push_back(s0.e);
+        flushing.push_back(s1.e);
+        done.count = flushing.size();
+        ucs_time_t loop_end_limit = ucs_get_time() + ucs_time_from_sec(50.0);
+        while (ucs_get_time() < loop_end_limit) {
+            std::list<entity *>::iterator i = flushing.begin();
+            while (i != flushing.end()) {
+                status = uct_ep_flush((*i)->ep(0), UCT_FLUSH_FLAG_CANCEL, &done);
+                if (status == UCS_ERR_NO_RESOURCE) {
+                    i++;
+                } else {
+                    ASSERT_UCS_OK_OR_INPROGRESS(status);
+                    i = flushing.erase(i);
+                    if (status == UCS_OK) {
+                        done.count--;
+                    }
+                }
+            }
+
+            if (flushing.empty()) {
+                break;
+            }
+
+            while (progress()) {}
+            usleep(100);
+        }
+        ASSERT_UCS_OK_OR_INPROGRESS(status);
+
+        if (done.count) {
+            wait_for_flag(&done.flag);
+        }
+
+        s1.e->destroy_eps();
+        s1.e->connect(0, *s0.e, 0);
+    }
+
+    typedef ucs_status_t (uct_cancel_test::* send_func_t)(side *s);
+
+    void fill(send_func_t send) {
+        ucs_status_t status;
+        std::list<side *> filling;
+
+        filling.push_back(&s0);
+        filling.push_back(&s1);
+        while (!filling.empty()) {
+            std::list<side *>::iterator i = filling.begin();
+            while (i != filling.end()) {
+                status = (this->*send)(*i);
+                if (status == UCS_ERR_NO_RESOURCE) {
+                    i = filling.erase(i);
+                } else {
+                    ASSERT_UCS_OK_OR_INPROGRESS(status);
+                    i++;
+                }
+            }
+        }
+    }
+};
+
+UCS_TEST_SKIP_COND_P(uct_cancel_test, am_zcopy,
+                     !check_caps(UCT_IFACE_FLAG_AM_ZCOPY)) {
+    const resource *r = dynamic_cast<const resource*>(GetParam());
+    int count = 100;
+
+    if (r->tl_name != "rc_mlx5" && r->tl_name != "rc_verbs") {
+        UCS_TEST_SKIP_R("not supported yet");
+    }
+
+    for (int i = 0; i < count; ++i) {
+        fill(&uct_cancel_test::am_zcopy);
+        flush_and_reconnect();
+    }
+}
+
+UCS_TEST_SKIP_COND_P(uct_cancel_test, am_bcopy,
+                     !check_caps(UCT_IFACE_FLAG_AM_ZCOPY)) {
+    const resource *r = dynamic_cast<const resource*>(GetParam());
+    int count = 100;
+
+    if (r->tl_name != "rc_mlx5" && r->tl_name != "rc_verbs") {
+        UCS_TEST_SKIP_R("not supported yet");
+    }
```

Comment:
add this to common function which is used by UCS_TEST_SKIP_COND_P

Reply:
we want to remove it i.e. make test pass for any transport, don't we?

Reply:
yes, but for some time we will just update the list of transports we support over time (e.g add ud, dc, etc)